pydantic/_internal/_std_types_schema.py

```diff
@@ -190,31 +191,31 @@ def uuid_schema(_schema_generator: GenerateSchema, uuid_type: type[UUID]) -> cor
 
 
 @schema_function(PurePath)
-def path_schema(_schema_generator: GenerateSchema, path_type: type[PurePath]) -> core_schema.LaxOrStrictSchema:
-    metadata = build_metadata_dict(js_override={'type': 'string', 'format': 'path'})
-    # TODO, is this actually faster than `function_after(...)` as above?
-    lax = core_schema.union_schema(
-        [
-            core_schema.is_instance_schema(path_type, json_types={'str'}, metadata=metadata),
-            core_schema.general_after_validator_function(
-                _validators.path_validator,
-                core_schema.str_schema(),
-                metadata=metadata,
-            ),
-        ],
-        custom_error_type='path_type',
-        custom_error_message='Input is not a valid path',
-        strict=True,
-    )
+@schema_function(PathLike)
+def path_schema(_schema_generator: GenerateSchema, path_type: type[PathLike]) -> core_schema.LaxOrStrictSchema:
+    construct_path = PurePath if path_type is PathLike else path_type
+
+    def path_validator(__input_value: str) -> PathLike:
+        try:
+            return construct_path(__input_value)  # type: ignore
+        except TypeError as e:
+            raise PydanticCustomError('path_type', 'Input is not a valid path') from e
+
+    instance_schema = core_schema.is_instance_schema(path_type, json_types={'str'}, json_function=path_validator)
 
     return core_schema.lax_or_strict_schema(
-        lax_schema=lax,
-        strict_schema=core_schema.general_after_validator_function(
-            lambda x, _: path_type(x),
-            core_schema.is_instance_schema(path_type, json_types={'str'}),
-            serialization=core_schema.to_string_ser_schema(),
-            metadata=metadata,
+        lax_schema=core_schema.union_schema(
+            [
+                instance_schema,
+                core_schema.no_info_after_validator_function(path_validator, core_schema.str_schema()),
+            ],
+            custom_error_type='path_type',
+            custom_error_message='Input is not a valid path',
+            strict=True,
         ),
+        strict_schema=instance_schema,
+        serialization=core_schema.to_string_ser_schema(),
+        metadata=build_metadata_dict(js_override={'type': 'string', 'format': 'path'}),
```

Comment:
I think `js_override` is not valid https://github.com/pydantic/pydantic/commit/c8b978ecd4136fa41baed58dee7d50cf6280cee5

Reply:
good point, fixed.